### DISCUSSION TITLE
Allow block logging plugins to log RestoreNature/ClaimExpiration tasks

### DISCUSSION THREAD

**Discussion by zedwick** [GriefPrevention/GriefPrevention#1230] [Ideas]
I would like to be able to see expired claim cleanup changes shown in my CoreProtect logs, as it is I will find what looks like a griefed path but find no logs. This is momentarily confusing for me, but can be much more disorientating for other moderators who are checking the logs who don't have knowledge of how GriefPrevention works. It would also be really great to be able to easily roll back some of the changes; really sucks having to clean up paths in towns constantly following claim expiration.

**Comment by RoboMWM** [comment]
Not sure of a good way to break blocks in such a way that will show up properly in any block logger, other than having to hook into each respective block logger. Either way, the restorenature feature won't be included in version 20, so that'll be someone else's job.

**Comment by zedwick** [comment]
Uhoh.

Could you please either delay Version 20 until after an earlier version has been released compatible with 1.12 or support 1.12 on <20 releases. Gutting the plugin and removing a lot of the functionality provided by Grief Prevention without giving time to find/create alternatives seems like a bad plan.

I understand why you would want to simplify and narrow the focus of the plugin, but we do rely on the existing feature set.

**Comment by RoboMWM** [comment]
hence version 20. The legacy version 16 branch will remain, and semi-maintained.

**Comment by zedwick** [comment]
Good to hear, thanks!

**Comment by TomLewis** [comment]
@RoboMWM I know its off topic, but I'm  out of the loop (Ive been away) whats happening to GriefPrevention? is it splitting up? If so where can I read about the changes?

**Comment by TomLewis** [comment]
I suggested years ago, that the worldedit //regen be used. it would only need to make a selection of the old claim with //pos1 and //pos2 then //regen, this all works from console already, and WE handles it all.

WorldEdit can easily be recorded into CoreProtect, I do it on my creative.

So implement a better regenerating based on existing tools that everyone already has AND it will allow you to log it.
